### PR TITLE
Add learn++ ensemble method and test

### PR DESCRIPTION
**Pull request by caoquan95**
Changes proposed in this pull request:
- Add learn++ ensemble algorithm
- Add test for learn++ ensemble algorithm

@scikit-multiflow/scikit-multiflow-repo-admins

Checklist

- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] Tests are included for new functionality.
- [x] Travis CI build passes.
- [x] Test Coverage is maintained (threshold 0.2%)

### CODE REVIEW

**Comment by jacobmontiel** [comment]
Hello @caoquan95 

It seems that your commit includes a lot of unrelated files. Here is an option to avoid polluting the project's history:

https://stackoverflow.com/a/51400593/4765776

**Note:** Keep a copy since they will be removed alongside the dropped commit. 
After removing the corrupted commit you can commit again you changes.

**Comment by caoquan95** [comment]
> Hello @caoquan95
> 
> It seems that your commit includes a lot of unrelated files. Here is an option to avoid polluting the project's history:
> 
> https://stackoverflow.com/a/51400593/4765776
> 
> **Note:** Keep a copy since they will be removed alongside the dropped commit.
> After removing the corrupted commit you can commit again you changes.

I removed all the unnecessary files and merge all the commits into one

**Comment by jacobmontiel** [comment]
Thanks @caoquan95 

We have a very good start point now. Here are my comments:

1. learn++ is a batch incremental method (your implementation sets the minimum batch size to 100). Thus we could define `window_size` as a parameter of this method (default is 100).

2. The `partial_fit` method assumes that the user will pass at least 100 samples. A more robust way is to keep a buffer of the train data and once the buffer is complete then we proceed with the training:

A similar aproach is available in the `BatchIncremental` method:
https://github.com/scikit-multiflow/scikit-multiflow/blob/1a513f147edc2900790874411d78f8ab6da715dd/src/skmultiflow/meta/batch_incremental.py#L47-L73

In order to reduce the impact on your code you can turn the current `partial_fit` method into a non-public method `_partial_fit()` and use the public method `partial_fit(...)` to manage the buffer and call `_partial_fit(...)` when necessary.

Let me know if this is clear.

**Comment by caoquan95** [comment]
@jacobmontiel 
Thank you very much. I understood the problem. I will fix it

**Comment by caoquan95** [comment]
@jacobmontiel  I modify the code as u said

File: tests/meta/test_learn++.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    expected_acc = 0.568
+    expected_predictions = [1, 0, 0, 1, 1, 0, 0, 0, 0, 1]
+
+    assert np.alltrue(predictions == expected_predictions)
+    assert np.isclose(expected_acc, acc)
+    assert corrects == expected_correct_predictions
+    assert type(learner.predict(X)) == np.ndarray
+
+
+test_learn_pp()
```
Comment (jacobmontiel) [inline tests/meta/test_learn++.py]: Not necessary

**Comment by jacobmontiel** [comment]
Thanks @caoquan95 

I have started the review and added some comments in the test file. From this point please add new commits normaly in order to keep track of the changes. They will be included in the pull request.

Also, please update the `__init__py` file in the `src/skmultiflow/meta` package to include `LearnPP`

**Comment by caoquan95** [comment]
@jacobmontiel 
I updated the __init__.py and the test_learn++.py file.

**Comment by jacobmontiel** [comment]
@caoquan95 
There is still the issue in the test file regarding `Pipeline`

**Comment by caoquan95** [comment]
Thanks, @jacobmontiel, What is the issue?

**Comment by jacobmontiel** [comment]
In the test file, it is not clear the purpose of lines 16-18. Furthermore, you don't need to wrap the classifier with `Pipeline`.

**Comment by jacobmontiel** [comment]
Thanks @caoquan95! 

I will merge this PR.

File: tests/meta/test_learn++.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    stream = RandomTreeGenerator(tree_random_state=2212, sample_random_state=2212)
+    stream.prepare_for_use()
+    estimator = DecisionTreeClassifier(random_state=2212)
+    classifier = LearnPP(base_estimator=estimator, n_estimators=5, n_ensembles=5, random_state=2212)
+
+    m = 200
+
+    learner = Pipeline([('classifier', classifier)])
+    X, y = stream.next_sample(m)
+    learner.partial_fit(X, y, classes=stream.target_values)
```
[the remark this answers is not in the document]
Comment (caoquan95) [reply tests/meta/test_learn++.py]: @jacobmontiel I didn't see these comments. I just saw them when you approved the PR. Where to find these comments?

[the remark this answers is not in the document]
Comment (jacobmontiel) [reply tests/meta/test_learn++.py]: When a review is created the reviewer can add comments directly in the code (as the ones above). You can either see them in the code view or listed in the PR history (attached to the review)

[the remark this answers is not in the document]
Comment (caoquan95) [reply tests/meta/test_learn++.py]: I found it. Thank you very much :dancing_men:

**Pull request merged**